.cirrus.yml

```diff
@@ -142,6 +142,8 @@ test_skopeo_task:
         "${SKOPEO_PATH}/${SCRIPT_BASE}/runner.sh" build
     validate_script: >-
         "${SKOPEO_PATH}/${SCRIPT_BASE}/runner.sh" validate
+    doccheck_script: >-
+        "${SKOPEO_PATH}/${SCRIPT_BASE}/runner.sh" doccheck
```

Comment:
Mentioned in a reply to a review comment (above), but might as well put it here.

I would prefer this be included in the 'validate' task higher up.  I *think* we can get away with dropping `validate_script` here (it's redundant).  This will also simplify the necessary copy/paste of this task over into c/image's `.cirrus.yml`.

Hopefully that makes sense.

Reply:
Alternatively, I'm also okay with having this be in a dedicated/standalone task.

Reply:
I agree `validate` in `test_skopeo_task` seems unnecessary (it is also not very related to the c/image tests, `validate` mostly targets the non-`vendor` parts of the codebase).

Moving the man page checks into `validate` might not be a good fit because the two checks depend on a built `bin/skopeo` binary, and at that point a separate `validate` “quick check” that is a prerequisite for other CI tasks loses some of its benefits. (That’s assuming that `validate_task` is intended to be fast and small, which is a guess, not a documented design.)

Reply:
Running in `validate` was my first approach; @mtrmac pushed against it as being confusing. Also, running in `validate` might make it run on Macs, which we 100% do not want.

Reply:
The thing I was worried about being confusing in this PR is `runner.sh validate` running both `validate-local` and `validate-docs`.

I think @cevich is suggesting moving this into `validate_task`, which does not involve `runner.sh` at all.

(AFAICS neither `validate_task` nor `test_skopeo_task` currently runs on Macs, though the current `validate_task`, before the dependency on recent `sed`, could in principle be moved to that platform.)

Reply:
It sounds to me like this should get it's own task then.  It could probably run in a container, but I'm okay if a VM is needed (note: this comes with a 1-minute spin-up time hit).  Something like (not tested):

```yaml
validate_task:
    only_if: $CIRRUS_PR != ''
    depends_on:
        - validate
    container:  *build_container
    script: |
        "${GOSRC}/${SCRIPT_BASE}/runner.sh" setup
        "${SKOPEO_PATH}/${SCRIPT_BASE}/runner.sh" build
        "${SKOPEO_PATH}/${SCRIPT_BASE}/runner.sh" doccheck
```

**note:** The `setup` step is probably only needed for the (accident prevention) `dnf erase skopeo`.  The default VM setup has 1-2gig of disk space available, which should be fine for this task.  I don't have actual knowledge of this though, so maybe `setup` is needed.

Reply:
If `validate_task` can run in a container, this one looks like it should run in a container similarly easily (same dependencies on Go to build the binary, + shell/perl to run the tests.)

(Does `runner.sh setup` work from inside a container? I’d expect the `growpart` to fail. Anyway the existing `validate_task` could be a starting point.)

Reply:
Right, in that case just `s/.*\/runner.sh" setup/dnf erase -y skopeo/` and it should be good to go.  Unless there are new perl requirements, that particular container image should already have what's needed.

Reply:
"new" meaning, something beyond what Podman and Buildah need for the same check.

Reply:
I'm completely lost here. IIUC you want a new `something_task` in `.cirrus.yml`? If so, I've just pushed a new change that (I think) does that. I'm not really the best at cirrus yaml, so I may have completely gotten it wrong.